Gems/Atom/RHI/Code/Source/RHI/DrawList.cpp

```diff
@@ -47,7 +47,11 @@ namespace AZ::RHI
                     {
                         return a.m_sortKey < b.m_sortKey;
                     }
-                    return a.m_depth > b.m_depth;
+                    if (a.m_depth != b.m_depth)
+                    {
+                        return a.m_depth > b.m_depth;
+                    }
+                    return a.m_item < b.m_item;
```

Comment:
This fix needs to be applied to all enums and not just KeyThenReverseDepth as this issue could happen for any of the other use cases.